misc/userscripts/format_json

```diff
@@ -19,29 +20,23 @@
 #
 # Bryan Gilbert, 2017
 
+# do not run pygmentize on files larger than this amount of bytes
+MAX_SIZE_PRETTIFY=10485760  # 10 MB
 # default style to monokai if none is provided
 STYLE=${1:-monokai}
-# format json using jq
-FORMATTED_JSON="$(cat "$QUTE_TEXT" | jq '.')"
 
-# if jq command failed or formatted json is empty, assume failure and terminate
-if [ $? -ne 0 ] || [ -z "$FORMATTED_JSON" ]; then
-    echo "Invalid json, aborting..."
-    exit 1
-fi
-
-# calculate the filesize of the json document
-FILE_SIZE=$(ls -s --block-size=1048576 "$QUTE_TEXT" | cut -d' ' -f1)
+TEMP_FILE="$(mktemp)"
+jq . "$QUTE_TEXT" >"$TEMP_FILE"
 
-# use pygments to pretty-up the json (syntax highlight) if file is less than 10MB
-if [ "$FILE_SIZE" -lt "10" ]; then
-    FORMATTED_JSON="$(echo "$FORMATTED_JSON" | pygmentize -l json -f html -O full,style=$STYLE)"
+# try GNU stat first and then OSX stat if the former fails
+FILE_SIZE=$(
+  stat --printf="%s" "$TEMP_FILE" 2>/dev/null ||
+  stat -f%z "$TEMP_FILE" 2>/dev/null
```

Comment:
It's probably a good idea to exit after this line just in case no `stat` is found.

Reply:
I do not think that's necessary since `set -e` is in effect.

Reply:
ah, my bad, ignore me :)